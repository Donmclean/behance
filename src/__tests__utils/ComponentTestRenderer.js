import React from 'react';
import { store, history } from '../js/utils/createStore';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import { ConnectedRouter } from 'react-router-redux';

const wrapComponent = (componentToRender) => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Route component={componentToRender} />
        </ConnectedRouter>
    </Provider>
);

export const getFullComponentJSON = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return renderer.create((componentWrapper).toJSON());
};

export const getFullComponentRenderWrapper = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return mount(componentWrapper);
};

export const getShallowComponentRendererWrapper = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return shallow(componentWrapper);
};