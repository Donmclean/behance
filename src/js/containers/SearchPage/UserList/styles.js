import styled from 'styled-components';

export const UserListWrapper = styled.div`
    position: relative;
    width: 100%;
    margin: 50px auto;
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: center;
`;

export const UserListItemWrapper = styled.div`
    padding: 10px;
    color: #ffffff;
    margin: 10px;
    text-align: center;
    font-size: 30px;
    text-decoration: none;
    width: 200px;
    box-shadow: 0px 4px 11px 0px rgba(0,0,0,0.75);
`;

export const UserListImageWrapper = styled.div`
    display: inline-block;
    cursor: pointer;
`;