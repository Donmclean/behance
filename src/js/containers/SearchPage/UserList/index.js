import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { UserListWrapper, UserListItemWrapper, UserListImageWrapper } from './styles'; 
import { BodyText, MissingImageBox } from '../../../themeComponents';
import { userProfileRoot } from '../../../constants/routes';

const UserList = ({ users = [] }) => (
    <UserListWrapper>
        {users.map((user, index) => (
            <Link key={index} to={`/${userProfileRoot}/${user.id}`}>
                <UserListItemWrapper>
                    <UserListImageWrapper>
                        {user.images ? (
                            <img src={user.images['138']}/>
                        ) : (
                            <MissingImageBox imageSize={138} />
                        )}
                    </UserListImageWrapper>
                    <BodyText>{user.display_name}</BodyText>
                </UserListItemWrapper>
            </Link>
        ))}
    </UserListWrapper>
);

UserList.propTypes = {
    users: PropTypes.array.isRequired
};

export default UserList;