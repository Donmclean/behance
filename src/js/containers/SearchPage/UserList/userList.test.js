import React from 'react';
import UserList from './index';

import { getFullComponentRenderWrapper } from '../../../../__tests__utils/ComponentTestRenderer';

describe('UserList Component', () => {
    it('renders <UserList /> component successfully in to the DOM with required props', () => {
        const props = {
            users: [{
                id: 1,
                display_name: 'testname',
                images: {
                    '138': 'img/url'
                }  
            }]
        };

        const UserListWithProps = () => (<UserList {...props} />);

        const tree = getFullComponentRenderWrapper(UserListWithProps);

        expect(tree.debug()).toMatchSnapshot();
    });
});