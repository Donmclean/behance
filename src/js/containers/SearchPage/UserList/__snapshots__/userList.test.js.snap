// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList Component renders <UserList /> component successfully in to the DOM with required props 1`] = `
"<Provider store={{...}}>
  <ConnectedRouter history={{...}}>
    <Router history={{...}}>
      <Route component={[Function: UserListWithProps]}>
        <UserListWithProps match={{...}} location={{...}} history={{...}} staticContext={[undefined]}>
          <UserList users={{...}}>
            <styled.div>
              <div className=\\"sc-bdVaJa Ewpce\\">
                <Link to=\\"/user/1\\" replace={false}>
                  <a onClick={[Function]} href=\\"/user/1\\">
                    <styled.div>
                      <div className=\\"sc-bwzfXH bkRIqz\\">
                        <styled.div>
                          <div className=\\"sc-htpNat eJimqb\\">
                            <img src=\\"img/url\\" />
                          </div>
                        </styled.div>
                        <styled.p>
                          <p className=\\"sc-bxivhb hTgdeh\\">
                            testname
                          </p>
                        </styled.p>
                      </div>
                    </styled.div>
                  </a>
                </Link>
              </div>
            </styled.div>
          </UserList>
        </UserListWithProps>
      </Route>
    </Router>
  </ConnectedRouter>
</Provider>"
`;
